src/kudu/tserver/tablet_server.cc

```diff
@@ -129,12 +130,35 @@ Status TabletServer::Init() {
   return Status::OK();
 }
 
-Status TabletServer::WaitInited() {
+void TabletServer::InitTabletManagerTask() {
+  Status s = InitTabletManager();
+  if (!s.ok()) {
+    LOG(ERROR) << "Unable to init master catalog manager: " << s.ToString();
+  }
+  init_status_.Set(s);
+}
+
+Status TabletServer::InitTabletManager() {
+  if (tablet_manager_->IsInitialized()) {
+    return Status::IllegalState("Catalog manager is already initialized");
+  }
+  RETURN_NOT_OK_PREPEND(tablet_manager_->Init(is_first_run_),
+                        "Unable to initialize catalog manager");
   return Status::OK();
-  //return tablet_manager_->WaitForAllBootstrapsToFinish();
+}
+
+Status TabletServer::WaitForTabletManagerInit() const {
+  return init_status_.Get();
 }
 
 Status TabletServer::Start() {
+  RETURN_NOT_OK(StartAsync());
```

Comment:
Why did we asyncify the initialization?

Reply:
Just followed the naming convention inside Kudu. The start function was always submitting the closure to init_pool.